portfolio/src/main/java/com/google/sps/servlets/PizzaSliceDataServlet.java

```diff
@@ -0,0 +1,104 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.data.User;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList; 
+import java.util.Arrays; 
+import java.util.List;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+
+@WebServlet("/pizza-data")
+public class PizzaSliceDataServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json");
+    Query query = new Query("Vote").addSort("slice", SortDirection.ASCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List<Integer> pizzaVotes = new ArrayList<>(Arrays.asList(0, 0, 0, 0, 0, 0));
+    for (Entity entity : results.asIterable()) {
+      Integer slice = Integer.parseInt(entity.getProperty("slice").toString());
+      if (slice > 5 || slice < 0) {
+        System.err.println("Slice number is not in [0, 5]: " + slice);
+        response.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        response.getWriter().println("Slice number should be in [0, 5]");
+        return;
```

Comment:
I would not respond with BAD_REQUEST here because the data is coming from internal storage.  BAD_REQUEST means that the client did something wrong which is not the case here.
Maybe better just do the error print and ignore this value.

Reply:
Wouldn't it be SC_INTERNAL_SERVER_ERROR in that case?